File: game/scripts/vscripts/items/item_vitality_custom/item_vitality_custom.lua
item_vitality_custom = class({})
LinkLuaModifier( "modifier_item_vitality_custom", "items/item_vitality_custom/modifier_item_vitality_custom", LUA_MODIFIER_MOTION_HORIZONTAL )

function item_vitality_custom:GetIntrinsicModifierName()
    return "modifier_item_vitality_custom"
end

File: src/vscripts/items/item_vitality_custom.ts
import { CustomAbility } from "../abilities/framework/custom_ability";
import { CustomModifier } from "../abilities/framework/custom_modifier";
import { registerAbility, registerModifier } from "../lib/dota_ts_adapter";

@registerAbility("item_vitality_custom")
class ItemVitalityBooster extends CustomAbility {
    GetIntrinsicModifierName() {
        return ModifierItemVitalityBooster.name;
    }
}

@registerModifier("modifier_item_vitality_custom")
class ModifierItemVitalityBooster extends CustomModifier {
    DeclareFunctions() {
        return [ModifierFunction.EXTRA_HEALTH_BONUS];
    }

    GetModifierExtraHealthBonus() {
        return this.Value("extra_health");
    }

    GetEffectName() {
        return "particles/vitality_effect/omniknight_repel_buff_ti8_glyph.vpcf";
    }

    GetEffectAttachType() {
        return ParticleAttachment.ABSORIGIN_FOLLOW;
    }
}
